### PR TITLE
[Gunveer | Harshit] Fix the postgres port issue while running docker.

### LINKED ISSUE #3: Can't connect to local postgres while running docker.
While MQL tool with docker, MQL is unable to connect to the database running in the local machine.

Steps to reproduce:
1. Clone the repo, and have your local postgres running.
2. Run make install.
3. In MQL dashboard try to connect to local database using 'Add Database'.

Expected behaviour:
Database connection should be successful. We should be able to ask natural language queries.

Actual behaviour:
Throws internal server error, and unable to connect to the local database.

Error Message:
Cannot assign the requested port. Happening because the local postgres and docker are trying to run on the same port.
And if the ports are right it gives error - 'Connection refused', because docker is trying to connect to the local database using the hostname as localhost.

### PR DESCRIPTION
**Pull request by gunveer-shuru**
1. Fix the port issue by changing the port to 5433 for docker.
2. Add a check if the 5433 port is free or not.
3. Update the readme file accordingly.
4. Add a hint in the UI on the 'Add Database' page suggesting the user to change the hostname if using docker.

resolves #3

### CODE REVIEW

**Comment by ashok-shuru** [comment]
Message to use `host.docker.internal` can be more clear. 
Like: If the database is running locally & MQL is running using docker then use 'host.docker.internal' instead of localhost.

**Comment by gunveer-shuru** [comment]
Description for the following commit: [Gunveer | Harshit] Fix tests, database names, and openAI client:

1. Change the database names from analytics and analytics_test to mql, and mql_test everywhere respectively. They were lingering in the project causing confusion if the user were to run the project on his local machine instead of docker.
2. Fix the tests for embedding changes.
3. Fix the tests for queries, they were missing the 'api/' in the path.
4. Fix the embeddings issue according to the new contract of openAI.
5. Fix the information text for the UI in 'Add Database', and in the readme as well.
6. Removed the exposed ports in the docker-compose.yml file as suggested by Harsh.

**Review by harshshurutech — CHANGES_REQUESTED** [review]
I tested the changes and encountered a login failure due to an `internal server error`.

Steps to reproduce:

1. Execute: `make install`
2. Log in using:
   - Username: `admin@example.com`
   - Password: `admin`

**Comment by gunveer-shuru** [comment]
Added another commit where we have removed the command check-free-port-for-postgres as it is not longer required because the port is no longer exposed

File: server/.env [hunk 1]
```diff
@@ -1,4 +1,4 @@
-DATABASE_URL="postgresql://shuru:password@postgres:5432/mql"
+DATABASE_URL="postgresql://shuru:password@localhost:5432/mql"
```
Comment (harshshurutech) [inline server/.env:1]: This should be `DATABASE_URL="postgresql://shuru:password@postgres:5432/mql"`

File: server/.env [hunk 2]
```diff
@@ -1,4 +1,4 @@
...
 TEST_DATABASE_URL="postgresql://shuru:password@localhost:5432/mql_test"
```
Comment (harshshurutech) [inline server/.env:2]: This should be `TEST_DATABASE_URL="postgresql://shuru:password@localhost:5432/mql_test"`

**Comment by gunveer-shuru** [comment]
Resolved the issue in env file, now using hostname postgres for running in docker, and localhost for running in local machine.

**Comment by harshshurutech** [comment]
LGTM

**Review by harshshurutech — APPROVED** [review]

**Comment by ashok-shuru** [comment]
In the env file, the hostname can be `postgres`, if a user wants to use docker we should keep minimum steps to run it. Right now in the case of docker also we need to add the hostname manually. So, as we mention in Readme if a user is running without docker then they should change it, we can use the default value for docker. 

Another way is to add a hostname in the env using Dockerfile

**Comment by gunveer-shuru** [comment]
> In the env file, the hostname can be `postgres`, if a user wants to use docker we should keep minimum steps to run it. Right now in the case of docker also we need to add the hostname manually. So, as we mention in Readme if a user is running without docker then they should change it, we can use the default value for docker.
> 
> Another way is to add a hostname in the env using Dockerfile

Not changing the information in the readme file, as over information is not bad, IMO.

Agreed with the default value 'postgres', as it would reduce the steps required to get MQL running.

**Comment by ashok-shuru** [comment]
1. No, That was a typo, It is already there so we should keep it as it is.
2. Okay, please let me know once you make these changes

**Pull request merged**